### PR TITLE
`install_sct.bat`: Use `requirements-freeze.txt` if present

### PR DESCRIPTION
**Pull request by joshuacwnewton**
#### GitHub

- [ ] I've given this PR a concise, self-descriptive, and meaningful title
- [ ] I've linked relevant issues in the PR body
- [ ] I've applied [the relevant labels](https://intranet.neuro.polymtl.ca/geek-tips/contributing#pr-labels-a-href-pr-labels-id-pr-labels-a) to this PR
- [ ] I've applied a [release milestone](https://github.com/spinalcordtoolbox/spinalcordtoolbox/milestones) (major, minor, patch) in line with [Semantic Versioning guidelines](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Misc%3A-Creating-a-new-release#convention-for-naming-releases) 
- [ ] I've assigned a reviewer

#### PR contents

- [ ] I've consulted [SCT's internal developer documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Tests) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

This is a bugfix for releases, which use `requirement-freeze.txt`, which is the output of `pip freeze` at time of release.

Without these changes, the more open `requirements.txt` would be installed instead of the frozen requirements.

## Linked issues

N/A.

### CODE REVIEW

File: install_sct.bat [hunk 1]
```diff
@@ -28,7 +28,11 @@ echo:
 echo ### Installing SCT and its dependencies...
 rem Pip needs to be upgraded because default p3.7 pip won't resolve dependency conflicts correctly
 python -m pip install --upgrade pip || goto error
-pip install -r requirements.txt || goto error
+if exist requirements-freeze.txt (
+  pip install -r requirements-freeze.txt || goto error
+) else (
+  pip install -r requirements.txt || goto error)
+)
```
Comment (RignonNoel) [inline install_sct.bat:35]: Maybe should we edit the `echo` to clarify the special case for support and debug purpose ?

```bash
echo ### Installing SCT and its dependencies... (using release frozed requirements)
```

Comment (RignonNoel) [reply install_sct.bat:35]: Maybe on two lines, allowing you to keep tje generic one outside the condition and adding the detail inside the condition of freeze.

Comment (joshuacwnewton) [reply install_sct.bat:35]: Ahh, good idea! :)

Comment (joshuacwnewton) [reply install_sct.bat]: I've done a barebones version of this in [737ee2e](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3749/commits/737ee2efcccd94b626c5d0e354ba1925a4beb2ee), but maybe we want to be more explicit by using plain English language (like you've done):

> ```
> echo ### Installing SCT and its dependencies... (using release frozed requirements)
> ```

**Review by RignonNoel — APPROVED** [review]

File: install_sct.bat (hunk 1 above)
Comment (RignonNoel) [reply install_sct.bat]: I think it's ok. People will understand the basics and we have the info for debugging.

**Pull request merged**